gateway/loaders/zookeeper/zookeeper.go

```diff
@@ -0,0 +1,289 @@
+package zookeeper
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-zookeeper/zk"
+	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ZookeeperClient struct {
+	acl  []zk.ACL
+	conn *zk.Conn
+}
+
+const TargetPath = "/targets"
+const RequestPath = "/request"
+
+type ConnectionConfig struct {
+	Addresses   []string          `yaml:"addresses"`
+	Request     string            `yaml:"request"`
+	Meta        map[string]string `yaml:"meta"`
+	Credentials CredentialsConfig `yaml:"credentials"`
+	TargetID    string
+}
+
+type CredentialsConfig struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+type RequestConfig struct {
+	Target    string   `yaml:"target"`
+	Paths     []string `yaml:"paths"`
+	RequestID string
+}
+
+// SetRequest will create a Zookeeper node in a cluster request tree.
```

Comment:
``SetRequest`` is unused and actually removed by a subsequent commit. I think we should just squash those 5 commits.